pdns/lua-recursor4.cc

```diff
@@ -1033,3 +1034,150 @@ void pdns_ffi_param_add_meta_single_int64_kv(pdns_ffi_param_t* ref, const char*
 {
   ref->params.meta[std::string(key)].intVal.insert(val);
 }
+
+struct pdns_postresolve_ffi_handle
+{
+public:
+  pdns_postresolve_ffi_handle(RecursorLua4::PostResolveFFIHandle& h) :
+    handle(h)
+  {
+  }
+  RecursorLua4::PostResolveFFIHandle& handle;
+  auto insert(std::string&& str)
+  {
+    const auto it = pool.insert(std::move(str)).first;
+    return it;
+  }
+
+private:
+  std::unordered_set<std::string> pool;
+};
+
+bool RecursorLua4::postresolve_ffi(RecursorLua4::PostResolveFFIHandle& h) const
+{
+  if (d_postresolve_ffi) {
+    pdns_postresolve_ffi_handle_t handle(h);
+
+    auto ret = d_postresolve_ffi(&handle);
+    return ret;
+  }
+  return false;
+}
+
+const char* pdns_postresolve_ffi_handle_get_qname(pdns_postresolve_ffi_handle_t* ref)
+{
+  auto str = ref->insert(ref->handle.d_dq.qname.toStringNoDot());
```

Comment:
The `toStringNoDot()` call used to be expensive, not sure if it was optimized since. The gettag_ffi API also has a raw API that returns the string in DNS wire format.

But we can keep it as is, as I don't think we will actually use this function?

Would a raw version even make sense in a response where I suppose label compression could be used?

Reply:
`ref->handle.d_dq.qname` is the original qname from the query, so we have the full raw name available.

Reply:
I think we always have the full name available at this point anyway, since we are using `DNSName` objects and not some view into a DNS packet?